### DISCUSSION TITLE
issue with TFY_API_KEY

### DISCUSSION THREAD

**Discussion by svolz-nova** [truefoundry/cognita#376] [General]
Hi, 
I am using aws ec2 instance with Amazon Linux, t2.xlarge, Docker version 25.0.5, Docker Compose version v2.29.7 
When simply following the setup for local deployment i get the error: `Exception: Value of TFY_API_KEY env var should be non-empty string
`
The faster-whisper container seems to be unhealthy. Is it correct that I need a tfy api key for local deployment? Or is there a comment missing somewhere? 
I didnt change anything in the models_config.yaml (except for using smaller 0.5b model) or the docker-compose.yaml 

Thanks so much for your help!

**Comment by FabrizioDG** [comment]
Hi,

I had this same issue. What I did was to create an account on truefoundry (using my company email, it cannot be a personal email) and I generated an API key and in compose.env I set the TFY_API_KEY and TFY_HOST where the host is something like https://<your_account_name>.truefoundry.cloud/

After that I was able to build with docker compose. I hope this helps.

**Comment by chiragjn** [comment]
Hello, you should not need a TFY_API_KEY unless you want to use TrueFoundry's AI gateway, As long as there are no entries in `models_config.yaml` that reference TFY_API_KEY it should not be required.

**Reply by chiragjn** [reply]
Replied here: https://github.com/truefoundry/cognita/issues/382#issuecomment-2447638256
We'll be fixing this soon

**Comment by mnvsk97** [comment]
Hello,
Apologies for the inconvenience. This issue has been fixed in #399. Please give it a try and let us know if you have problems. Thanks!
